src/core/view/CompassView.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Xournal++
+ *
+ * A Compass view
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <cairo.h>  // for cairo_t
+
+#include "model/Compass.h"  // for Compass
+#include "util/DispatchPool.h"
+#include "view/overlays/BaseStrokeToolView.h"
+
+#include "GeometryToolView.h"
+
+class Stroke;
+class OverlayBase;
+class ZoomControl;
+
+/**
+ * @brief A class that renders a compass
+ *
+ * The compass has vertical, horizontal and angular marks.
+ * The intersection angle with the x-axis is displayed as well.
+ *
+ * See the file development/documentation/Compass_Readme.md for
+ * details how the Compass is rendered.
+ *
+ * A temporary stroke is displayed when it is created near the
+ * outline of the Compass or at the marked radius (with angle 0).
+ */
+
+namespace xoj::view {
+class Repaintable;
+
+class CompassView: public GeometryToolView {
+
+public:
+    CompassView(const Compass* s, Repaintable* parent, ZoomControl* zoomControl);
+    ~CompassView() override;
+
+    void on(FlagDirtyRegionRequest, const Range& rg) override;
+    void on(UpdateValuesRequest, double h, double rot, cairo_matrix_t m) override;
+    void deleteOn(FinalizationRequest, const Range& rg) override;
+
+private:
+    void drawGeometryTool(cairo_t* cr) const override;
+    void drawDisplays(cairo_t* cr) const override;
+
+    double height = Compass::INITIAL_HEIGHT;
+    double rotation = 0.;
+    cairo_matrix_t matrix{CM, 0., 0., CM, Compass::INITIAL_X, Compass::INITIAL_Y};
+
+    /**
+     * @brief the radius of the semi-circle for the angular marks
+     */
+    double radius = 4.5;
```

Comment:
nit: These seem to be constants and should be marked as `const` or `constexpr`.

Reply:
They are not constant, but modified here:
https://github.com/rolandlo/xournalpp/blob/bfe1bd9d98e4612702f4e1d9efc3b17788713005/src/core/view/CompassView.cpp#L34-L47